playwright/test/helpers/search.ts

```diff
@@ -1,59 +1,56 @@
 import { Page } from 'playwright';
 import { expect } from '@playwright/test';
 import { isMobile } from '../contexts';
-import { formatFilterMobileButton } from '../selectors/search';
-
-export const worksSearchForm = '#search-searchbar';
 
 export const searchQueryAndSubmit = async (
   query: string,
   page: Page
 ): Promise<void> => {
-  await page.fill(worksSearchForm, query);
-  await page.press(worksSearchForm, 'Enter');
+  const searchBar = page.getByLabel('Search the catalogue');
+
+  await searchBar.fill(query);
+  await searchBar.press('Enter');
 };
 
-export const openFilterDropdown = async (id: string, page: Page) => {
+export const openFilterDropdown = async (name: string, page: Page) => {
   if (isMobile(page)) {
-    await page.click(formatFilterMobileButton);
+    await page.getByRole('button', { name: 'Filters' }).click();
   } else {
-    await page.click(`button[aria-controls="${id}"]`);
+    await page.getByRole('button', { name }).click();
   }
 };
 
-export const selectFilterAndWaitForApplied = async (
-  filterId: string,
+export const selectAndWaitForFilter = async (
+  filterName: string,
   checkboxId: string,
   page: Page
 ) => {
-  await openFilterDropdown(filterId, page);
+  await openFilterDropdown(filterName, page);
 
   const checkbox = page.locator(
-    `input[form="search-page-form"][name="${filterId}"][value="${checkboxId}"]`
+    `input[form="search-page-form"][value="${checkboxId}"]`
```

Comment:
This could probably be better but I'm not sure what else to use as we can't guarantee that the label text will be unique and it has to stay non-exact as the exact version would contain the number of results which is likely to change often.